test/functional/treesitter/inspect_tree_spec.lua

```diff
@@ -114,4 +114,56 @@ describe('vim.treesitter.inspect_tree', function()
             (fenced_code_block_delimiter)))) ; [2, 0] - [2, 3] markdown
       ]]
   end)
+
+  it('updates window if buffer is still open', function()
+    insert([[
+      print()
+      ]])
+
+    -- setup two windows for the same buffer
+    exec_lua([[
+      win = vim.api.nvim_get_current_win()
+      vim.api.nvim_open_win(0, false, {
+        win = 0,
+        split = 'left'
+      })
+    ]])
+
+    exec_lua([[
+      vim.treesitter.start(0, 'lua')
+      vim.treesitter.inspect_tree()
+    ]])
+
+    -- close initial window
+    exec_lua('vim.api.nvim_win_close(win, false)')
+
+    -- try to navigate to the remaining window
+    feed('<CR>')
+    eq('', n.api.nvim_get_vvar('errmsg'))
+  end)
+
+  it('updates inspect tree window', function()
```

Comment:
would it save a lot of code if this test is merged into the one above?

Reply:
Only managed to reduce 9 lines.